sdk/core/core-rest-pipeline/src/util/tokenCycler.ts

```diff
@@ -128,18 +139,20 @@ export function createTokenCycler(
       return refreshWorker !== null;
     },
     /**
-     * Produces true if the cycler SHOULD refresh (we are within the refresh
-     * window and not already refreshing)
+     * Produces true if the cycler SHOULD refresh.
+     * If tokens have a `refreshOn` property, tokens should refresh if that unix date has been reached,
+     * otherwise they should refresh if they are within the refresing window and not already refreshing.
      */
     get shouldRefresh(): boolean {
       return (
         !cycler.isRefreshing &&
-        (token?.expiresOnTimestamp ?? 0) - options.refreshWindowInMs < Date.now()
+        // If we received a "refreshOn" value from a token, and the "refreshOn" date has been reached, we refresh.
+        ((token?.refreshOn ? Date.now() > token.refreshOn : false) ||
```

Comment:
this expression is too complicated. Please pull out some parts into variables or something to make it more readable.